Subscription NB tests sometimes hang or produce a panic

**Describe the bug**
The NB subscribe tests sometimes hang or cause a panic.

**To Reproduce**
I have seen the panics only when run inside of a docker container. I see the hangs in both local and docker builds.

This is one of the panics:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x900ba4]

goroutine 72 [running]:
github.com/onosproject/onos-config/pkg/northbound/gnmi.listenForUpdates(0xc00006b260, 0xb5ae20, 0xc00021c5a0, 0xf7c720, 0xc00029d410, 0xc00029d3e0, 0xc00006b320)
	/go/src/github.com/onosproject/onos-config/pkg/northbound/gnmi/subscribe.go:168 +0x4f4
created by github.com/onosproject/onos-config/pkg/northbound/gnmi.listenOnChannel
	/go/src/github.com/onosproject/onos-config/pkg/northbound/gnmi/subscribe.go:128 +0x27a
FAIL	github.com/onosproject/onos-config/pkg/northbound/gnmi	0.047s

--- a/pkg/northbound/gnmi/subscribe_test.go
+++ b/pkg/northbound/gnmi/subscribe_test.go
@@ -282,7 +282,6 @@ func Test_Poll(t *testing.T) {
 		PollRequest: pollrequest,
 	}
 
-	//TODO need to block
 	go func() {
 		err = server.Subscribe(serverFake)
 		assert.NilError(t, err, "Unexpected error doing Subscribe")
@@ -299,9 +298,9 @@ func Test_Poll(t *testing.T) {
 	path3Poll := "leaf2a"
 	value := "13"
 
-	for i := 1; i < 4; i++ {
+	for i := 0; i < 4; i++ {
 		response := <-responsesChan
-		log.Info("response", response)
+		log.Info("Response ", response)
 		if len(response.GetUpdate().GetUpdate()) != 0 {
 			assertUpdateResponse(t, response, device1, path1Poll, path2Poll, path3Poll, value)
 		} else {

